### PR TITLE
CLN/INT: Rename _possibly to _maybe (GH15764)

### LINKED ISSUE #15764: CLN/INT: rename *possibly* -> *maybe* functions
we have a mixed bag on naming this. Let's change ``possibly`` -> ``maybe``. Keep everything else the same. (if you do the same search for ``maybe`` you will fine it to be far more common). Note that the *usage* of these functions will have to be renamed as well :>

```
(pandas) bash-3.2$ grep --include '*.py'  -R possibly pandas |grep def
pandas/computation/expr.py:    def _possibly_transform_eq_ne(self, node, left=None, right=None):
pandas/computation/expr.py:    def _possibly_downcast_constants(self, left, right):
pandas/computation/expr.py:    def _possibly_eval(self, binop, eval_in_python):
pandas/computation/expr.py:    def _possibly_evaluate_binop(self, op, op_class, lhs, rhs,
pandas/core/internals.py:def _possibly_compare(a, b, op):
pandas/indexes/base.py:    def _possibly_promote(self, other):
pandas/tests/types/test_cast.py:    def test_possibly_convert_objects_copy(self):
pandas/tseries/index.py:    def _possibly_promote(self, other):
pandas/tseries/tdi.py:    def _possibly_promote(self, other):
pandas/types/cast.py:def _possibly_convert_platform(values):
pandas/types/cast.py:def _possibly_downcast_to_dtype(result, dtype):
pandas/types/cast.py:def _possibly_cast_item(obj, item, dtype):
pandas/types/cast.py:def _possibly_convert_objects(values, convert_dates=True, convert_numeric=True,
pandas/types/cast.py:def _possibly_castable(arr):
pandas/types/cast.py:def _possibly_infer_to_datetimelike(value, convert_dates=False):
pandas/types/cast.py:def _possibly_cast_to_datetime(value, dtype, errors='raise'):
```

### PR DESCRIPTION
**Pull request by StanczakDominik**
Also rename "private" functions in pandas.type.cast

 - [x] closes #15764
 - [x] tests passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``

### CODE REVIEW

**Comment by jreback** [comment]
thanks @StanczakDominik exactly what I was looking.

ping on green. (hopefully linting passes too!).

**Comment by jreback** [comment]
couple of linting errors.


```
pandas/computation/expr.py:386:56: E127 continuation line over-indented for visual indent
pandas/core/internals.py:5144:80: E501 line too long (88 > 79 characters)
```

File: pandas/tests/types/test_cast.py [hunk 1]
```diff
@@ -9,33 +9,33 @@
...
+                               maybe_convert_string_to_object,
+                               maybe_convert_scalar,
+                               find_common_type)
 from pandas.types.dtypes import (CategoricalDtype,
                                  DatetimeTZDtype, PeriodDtype)
 from pandas.util import testing as tm
 
 
-class TestPossiblyDowncast(tm.TestCase):
+class TestMaybeDowncast(tm.TestCase):
```
Comment (StanczakDominik) [inline pandas/tests/types/test_cast.py]: I wasn't quite sure whether or not I should change the name of this class. It seemed logical as it seemed to refer to the "private" method I had changed.

**Comment by StanczakDominik** [comment]
Well, I guess I must have missed them somehow.

I'm not entirely sure why Codecov claims this would decrease coverage. It might be related to my comment above.

By the way, as this is my first foray into giving back to the awesome packages in the Python ecosystem, I'm not entirely sure if my thinking is correct: once test pass, I should squish my commits into one for inclusion in master, right?

File: pandas/tests/types/test_cast.py (hunk 1 above)
Comment (jreback) [inline pandas/tests/types/test_cast.py]: yes that is fine

**Comment by jreback** [comment]
no worries @StanczakDominik for future reference: http://pandas-docs.github.io/pandas-docs-travis/contributing.html#python-pep8

**Comment by jreback** [comment]
thanks @StanczakDominik

**Pull request closed without merging**